### ISSUE TITLE
.requiresArg doesn't work well with .string

### ISSUE THREAD

**Issue by TimothyGu** [bcoe/yargs#133]
```js
// test1.js
console.log(require('yargs')
  .describe('j', 'test')
  .requiresArg('j')
  .argv)
```

```js
// test2.js
console.log(require('yargs')
  .describe('j', 'test')
  .string('j')
  .requiresArg('j')
  .argv)
```

```sh
# good
$ node test1.js
Options:
  -j  test                                                                      

Missing argument value: j
# bad
$ node test2.js
{ _: [],
  j: '',
  '$0': '../../.nvm/versions/io.js/v1.6.3/bin/iojs testcase.js' }
```

**Comment by bcoe** [comment]
@TimothyGu this should now be fixed in `yargs@3.7.0`,

`npm install yargs@3.7.0 --save`.

**Comment by TimothyGu** [comment]
ok thanks, although it doesn't really matter for me anymore as I switched to commander a few days ago.
